### DISCUSSION TITLE
Record classes dont inherit ToString method from base.

### DISCUSSION THREAD

**Discussion by nathan130200** [dotnet/runtime#109484] [Q&A] [answered]
I am creating a node structure in XML for the XMPP protocol. To simplify the code structure regarding cloning XML elements to modify without changing the original. I used record classes and inheritance between record and the subtypes of elements.

However, the base record class that implements ToString generates the XML without indentation, but the classes inherited from records do not inherit the ToString from the record base class.

Is this behavior really intentional? Because I really like using the `with` operator with records, it's also the only reason why I turned them into record classes.

```cs
    public override string ToString() => ToString(false);


    public string ToString(bool indented)
    {
        var sb = new StringBuilder();


        var formatting = XmlFormatting.Default;


        if (indented)
            formatting |= XmlFormatting.Indented;


        using (var sw = new StringWriter(sb))
            WriteTo(sw, formatting);


        return sb.ToString();
    }
```

To get correct XML string representation of my Element class, i must override ToString again for each record class that inherits from my Element base class.

```cs
public sealed record StreamFeatures : Element
{
    public override string ToString() => base.ToString(false); // without override this method, it prints original record ToString instead Element::ToString()


    public StreamFeatures(StreamFeatures other) : base(other)
    {


    }


    public StreamFeatures() : base("stream:features", Namespaces.Stream)
    {


    }
```

**Comment by huoyaoyuan** [comment, marked as answer]
See https://github.com/dotnet/sdk/issues/43617 .
Record class always generates a new override for `ToString`, which uses the default record semantics and doesn't uses `base.ToString`.

> it prints original record ToString instead Element::ToString()

Note that there isn't one implementation of "record ToString". Every record generates its own ToString with that semantic.

**Reply by nathan130200** [reply]
Oh, thanks for the information. Unfortunately I ended up having to rollback the code to stop using records because of this. I had no idea it was possible to seal an override 🤯

**Reply by colejohnson66** [reply]
I think it might be possible to forward the ToString to the base implementation:

```csharp
public record MyRecord : MyRecordBase
{
    public override string ToString() =>
        base.ToString();
}
```

**Reply by jilles-sg** [reply]
In C# 10 and newer it is possible to seal a record's `ToString` implementation: https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-10#record-types-can-seal-tostring

Would that work?

**Reply by nathan130200** [reply]
@jilles-sg Yeah seal to string method should work.
